cfpb/cfgov-refresh
# College costs: templates and styles

Adds initial Jinja and LESS files for college costs app (the Javascript that runs the app will follow in future PRs)

## Additions

- Jinja template for each screen of the app
- College cost specific jinja macros: inline label/input, inline label/text, number callout, and customized expandable & secondary nav (It seemed preferable to create new macros for the expandable and nav rather than try to add options for the college cost specific features to the existing ones)
- LESS files
- Wagtail fields for intro screen of app

## Changes

- Updates radio button macro to make `class` parameter optional
- Updates notification macro to make `message` optional, change `explanation` element to div so it can contain lists, and add two new college cost specific variations (`money tips`, with green background and dollar icon, and `caution`, with red background and information icon). 

## Testing

1. Check out branch
2. `python cfgov/manage.py migrate`
3. `./frontend.sh`
4. In Wagtail admin, create a new page using the "College costs" page type
5. Verify that the intro screen is editable in Wagtail by entering content in the `Header` and/or `Content` field on the new page and checking that it appears above the "Get started" button in preview
5. Verify that the general layout on mobile and desktop matches the screenshots below. None of the interactions will be functional since this PR does not include Javascript, but all the app's screens should be visible
6. Verify the mobile vs desktop experience
    - Secondary nav fits our pattern on mobile (it will appear expanded sans JS)
    - `Costs not covered` sticky sidebar only appears on desktop and `Costs not covered` notifications only appear on mobile (on screens 3, `Grants and scholarships`, through 7,  `Other resources`) 


## Screenshots
### New or updated patterns
#### Money tip notification
<img width="596" alt="Screen Shot 2020-05-22 at 12 30 59 PM" src="https://user-images.githubusercontent.com/778171/82703093-231dc580-9c28-11ea-92da-c4854be75d96.png">

#### Caution notification

<img width="601" alt="Screen Shot 2020-05-22 at 12 30 47 PM" src="https://user-images.githubusercontent.com/778171/82703095-244ef280-9c28-11ea-8074-b3649f3bf00b.png">

#### Financial item expandable header with edit cue

<img width="599" alt="Screen Shot 2020-05-22 at 12 30 32 PM" src="https://user-images.githubusercontent.com/778171/82703096-24e78900-9c28-11ea-9e5d-b35bb5a80317.png">

#### Financial item input
<img width="591" alt="Screen Shot 2020-05-22 at 12 29 13 PM" src="https://user-images.githubusercontent.com/778171/82703097-25801f80-9c28-11ea-9c67-08734c683f64.png">

#### Financial item text
<img width="590" alt="Screen Shot 2020-05-22 at 12 29 04 PM" src="https://user-images.githubusercontent.com/778171/82703099-25801f80-9c28-11ea-8a99-14ee787d041f.png">

#### Number callout
<img width="581" alt="Screen Shot 2020-05-22 at 12 28 47 PM" src="https://user-images.githubusercontent.com/778171/82703101-25801f80-9c28-11ea-8811-3eb1057cd1b2.png">

### Mobile vs desktop
#### College costs secondary nav on desktop

<img width="302" alt="Screen Shot 2020-05-22 at 12 46 49 PM" src="https://user-images.githubusercontent.com/778171/82704416-52353680-9c2a-11ea-91fe-f318d3872f5b.png">

#### College costs secondary nav on mobile

<img width="494" alt="Screen Shot 2020-05-26 at 1 30 46 PM" src="https://user-images.githubusercontent.com/778171/82947623-33e67800-9f55-11ea-8615-920ed55fefc3.png">

#### Costs not covered sticky on desktop

<img width="322" alt="Screen Shot 2020-05-26 at 1 31 09 PM" src="https://user-images.githubusercontent.com/778171/82947698-495ba200-9f55-11ea-83b4-1f7a25b6f72b.png">

#### Costs not covered notification on mobile 

<img width="475" alt="Screen Shot 2020-05-26 at 1 31 00 PM" src="https://user-images.githubusercontent.com/778171/82947708-4c569280-9f55-11ea-8a3d-c4fb0311c33c.png">


#### College costs app, desktop (partial):
![desktop](https://user-images.githubusercontent.com/778171/82946815-ddc50500-9f53-11ea-921d-6b656b4bbc4a.png)

#### College costs app, mobile (partial):
![mobile](https://user-images.githubusercontent.com/778171/82946821-e1f12280-9f53-11ea-83aa-e9fb3dd08444.png)




## Notes
- It may be possible in a future PR to generalize some of the college costs macros so they can be used in other contexts (ie, `inline input` instead of `financial item input`)

## Todos

-

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
